static/source.js

```diff
@@ -0,0 +1,34 @@
+(function() {
+    function showSourceFiles(button, sideMenu, sourceCode) {
+        button.getElementsByClassName("pure-menu-link")[0].childNodes[0].textContent = "< ";
```

Comment:
Hmm, could you maybe add a title-text of "collapse" (and "expand" for the other)? This seems not great for accessibility.

Actually, @ahicks92 do you know how this is normally done for screen-readers? I assume this will read out as "left arrow" which doesn't seem ideal.

Reply:
I can't figure out how to get a test link out of the pr; is there a way to do that?

Reply:
@ahicks92: I can't host this PR so not possible unfortunately. :-/

@jyn514: Adding a title attribute would fix this issue I guess.

Reply:
I think you want  `aria-label`.  You'd also want [aria-expanded](https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaExpanded) I think.  Unfortunately this is tentative: let's say that I'm 70% confident.  Fortunately, I think most screen reader users would be able to figure this out even without changes, though improving this is definitely a good idea.

Not 100% on whether or not `title` is relevant, but probably not the end of the world to add one.  I think that title ends up being togglable and counted as a tooltip or something like that, not entirely replacing the text of the button in all contexts.

Reply:
As a followup to that note that `aria-expanded` if done right is going to make it obvious that the button expands or collapses something, and the common mistake is working the action into the label (e.g. "expand table of contents" as the label becomes "expand table of contents button collapsed" for example, which is irrelevant info).

Reply:
It's currently being hosted on my dev instance, <https://docs.rs.dev.nemo157.com/crate/tokio-util/0.6.7/source/src/codec/decoder.rs>.

Reply:
That link is dead, maybe a VPN or something.  In any case we still probably want changes here, so I'll check back in tomorrow.

Reply:
Ah, it's IPv6 only. I can maybe throw cloudflare in front of it to get an IPv4 reverse proxy :thinking:.

Reply:
Using cloudflare as a reverse proxy is a little more annoying than I expected, but hopefully this works now: <https://ipvold-docs-rs-dev.nemo157.com/crate/tokio-util/0.6.7/source/src/codec/decoder.rs>

Reply:
@ahicks92: Let's go for `aria-label`, `title` and `aria-expanded`. :)

Reply:
That's not a button tag. Missed that in my review somehow.  The labeling stuff only works on items of the right role.  The easiest (read: bug-free) way here is to use a button tag outside the list instead, but if we're willing to possibly accept bugs on some platforms then adding `role="button"` might or might not work.  Unfortunately screen readers get buggy around edge cases, and "li tagged as button inside a list" seems like one. I hacked it in via the inspector with Firefox+NVDA and it seems to work for me.  If there's  a reason we need to keep it this way rather than using a proper button we can try it and I'll do some more testing on the combinations I have access to.

note also that non-button buttons etc. like this need tabindex=0 for keyboard accessibility; you can't tab to them  by default either.

Reply:
I plan to use a `button`, for now it was mostly validating the UI (but I think it's ok now so I'll update it).